### PR TITLE
Self-signed certificates support for registry library

### PR DESCRIPTION
**Pull request by GeekArthur**
Signed-off-by: Jingfu Wang <jingfwan@redhat.com>

**Please specify the area for this PR**
Registry library

**What does does this PR do / why we need it**:
This PR add support for self-signed certificates to skip the TLS verification for self-signed certificates.

**Which issue(s) this PR fixes**:

Fixes https://github.com/devfile/api/issues/388

**PR acceptance criteria**:

- [x] Test (WIP) 

- [x] Documentation (WIP)

**How to test changes / Special notes to the reviewer**:
Build the registry CLI for testing.

### CODE REVIEW

**Review by johnmcollier — CHANGES_REQUESTED** [review]
Just a couple nits about the parameter name

We should also mention this on the cabal call on Thurs.

File: registry-library/cmd/root.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ var (
 	allResources bool
 	destDir      string
 	devfileType  string
+	insecure     bool
```
[the remark this answers is not in the document]
Comment (GeekArthur) [reply registry-library/cmd/root.go]: Done

File: registry-library/library/library.go [hunk 2]
```diff
@@ -63,7 +64,7 @@ type Registry struct {
 // GetRegistryIndex returns the list of stacks and/or samples, more specifically
 // it gets the stacks and/or samples content of the index of the specified registry
 // for listing the stacks and/or samples
-func GetRegistryIndex(registryURL string, devfileTypes ...indexSchema.DevfileType) ([]indexSchema.Schema, error) {
+func GetRegistryIndex(registryURL string, insecure bool, devfileTypes ...indexSchema.DevfileType) ([]indexSchema.Schema, error) {
```
[the remark this answers is not in the document]
Comment (GeekArthur) [reply registry-library/library/library.go]: Done

**Review by johnmcollier — APPROVED** [review]
We should

**Pull request merged**